File: src/main/java/com/hy/http/file/Binary.java
package com.hy.http.file;

import java.io.File;

public interface Binary {

    File getFile();

    /**
     * Length of byteArray.
     *
     * @return Long length.
     */
    long getLength();

    /**
     * Return the fileName, Can be null.
     *
     * @return File name.
     */
    String getFileName();

    /**
     * Return mimeType of binary.
     *
     * @return MimeType.
     */
    String getMimeType();
}

File: src/main/java/com/hy/http/file/Binary.kt
package com.hy.http.file

import java.io.File

interface Binary {

    val file: File

    /**
     * Length of byteArray.
     * @return Long length.
     */
    val length: Long

    /**
     * Return the fileName, Can be null.
     * @return File name.
     */
    val fileName: String

    /**
     * Return mimeType of binary.
     * @return MimeType.
     */
    val mimeType: String
}